python/ngen_conf/src/ngen/config/path_pair/typing.py

```diff
@@ -0,0 +1,8 @@
+from pathlib import Path
+from typing import Union, TypeVar
+from typing_extensions import TypeAlias
+
+T = TypeVar("T", bound=object, covariant=True)
```

Comment:
a little explanation as to what the difference in T and S are would be useful here

Reply:
done